### PR TITLE
Cunygb

### PR DESCRIPTION
**Pull request by aorinevo**

### CODE REVIEW

**Comment by aorinevo** [comment]
Merge cunygb down into master.

**Comment by aorinevo** [comment]
Hi Geoff, can you give me some feedback on what I have developed so far?

File: lib/WeBWorK/ContentGenerator/Instructor/GradeBook.pm [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+			CGI::td({-class=>"column-name"},'Student'),
+			CGI::td({-class=>"column-set",-valign=>'top'}, [@setLinks]) 
+		),
+		@studentLinks,
+		CGI::end_table(),
+	);
+	
+}
+
+#############################################################
```
Comment (aorinevo) [inline lib/WeBWorK/ContentGenerator/Instructor/GradeBook.pm]: @goehle, @mgage: I have seen this Grading Utility code in several places.  Should we move this into a separate module?

File: lib/WeBWorK/URLPath.pm [hunk 2]
```diff
@@ -778,6 +780,36 @@ our %pathTypes = (
...
+	instructor_gradebook => {
+		name    => 'GradeBook',
+		parent  => 'instructor_tools',
+		kids    => [ qw/instructor_gradebook_set_progress instructor_gradebook_user_progress/ ],
+		match   => qr|^gradebook/|,
+		capture => [ qw// ],
+		produce => 'gradebook/',
+		display => 'WeBWorK::ContentGenerator::Instructor::GradeBook',
+	},
+	instructor_gradebook_set_progress => {
```
Comment (aorinevo) [inline lib/WeBWorK/URLPath.pm]: This is not used and should be removed.

File: lib/WeBWorK/URLPath.pm [hunk 3]
```diff
@@ -778,6 +780,36 @@ our %pathTypes = (
...
+		name    => 'GradeBook',
+		parent  => 'instructor_gradebook',
+		kids    => [ qw// ],
+		match   => qr|^(set)/([^/]+)/|,
+		capture => [ qw/statType setID/ ],
+		produce => 'set/$setID/',
+		display => 'WeBWorK::ContentGenerator::Instructor::GradeBook',
+	},
+	instructor_gradebook_user_progress => {
```
Comment (aorinevo) [inline lib/WeBWorK/URLPath.pm]: This is not used and should be removed.

File: conf/defaults.config [hunk 4]
```diff
@@ -30,9 +30,10 @@
 
 include("conf/site.conf");
 include("VERSION");  # get WW version 
-include("PG_VERSION"); # get PG VERSION -- # these addresses are fragile.  
+#include("PG_VERSION"); # get PG VERSION -- # these addresses are fragile.  
                                            # Should we change the way include() works?
-
+$PG_VERSION  ='2.12';
```
Comment (aorinevo) [inline conf/defaults.config]: This is not going to be commited in the end.  The way the PG version is set breaks my local.  This is a hack to get my local to work.

**Comment by goehle** [comment]
Hi Aori,  

Unfortunately I'm pretty heavily engaged with some other projects at the moment so I won't be able to look at this pull request for a few weeks at least.  The only piece of advice I can give right now is that this pull request is much too complex to be pulled into the `master` branch.   You will need to rebase your changes onto `develop` and create a new pull request to `develop` before this feature can be vetted. 

Cheers.
Geoff.

**Pull request closed without merging**